src/main/java/com/nike/cerberus/store/ConfigStore.java

```diff
@@ -137,66 +137,14 @@ public ConfigStore(final AmazonS3 s3Client,
     }
 
     /**
-     * Writes the AZs to the configuration store.  This is a one time operation.  Subsequent calls will result in
-     * an {@link IllegalStateException} being thrown.
-     *
-     * @param az1 AZ 1
-     * @param az2 AZ 2
-     * @param az3 AZ 3
-     */
-    public void storeAzs(final String az1, final String az2, final String az3) {
-        synchronized (envDataLock) {
-            final Environment environment = getEnvironmentData();
-
-            if (StringUtils.isNotBlank(environment.getAz1())
-                    || StringUtils.isNotBlank(environment.getAz2())
-                    || StringUtils.isNotBlank(environment.getAz3())) {
-                throw new IllegalStateException("AZs are already defined for this environment!  Aborting save...");
-            }
-
-            environment.setAz1(az1);
-            environment.setAz2(az2);
-            environment.setAz3(az3);
-            saveEnvironmentData(environment);
-        }
-    }
-
-    /**
-     * Stores the stack ID for a specific component.
-     *
-     * @param stack Stack component
-     * @param stackId   Stack ID
-     */
-    public void storeStackId(final Stack stack, final String stackId) {
-        synchronized (envDataLock) {
-            final Environment environment = getEnvironmentData();
-            environment.getStackMap().put(stack, stackId);
-            saveEnvironmentData(environment);
-        }
-    }
-
-    /**
-     * Get the specific stack ID by component name.
-     *
-     * @param stack Stack name
-     * @return Stack ID
-     */
-    public String getStackId(final Stack stack) {
-        synchronized (envDataLock) {
-            final Environment environment = getEnvironmentData();
-            return environment.getStackMap().get(stack);
-        }
-    }
-
-    /**
      * Gets the server certificate name from the config store.
      *
      * @param stack Stack name
      */
     public String getServerCertificateName(final Stack stack) {
         synchronized (envDataLock) {
             final Environment environment = getEnvironmentData();
-            return environment.getServerCertificateIdMap().get(stack);
+            return environment.getServerCertificateIdMap().get(stack.getName());
```

Comment:
If the map was left as `Map<Stack,String>` then this could just be `get(stack)`.  Is that better?

Reply:
Stack doesn't serialize in a good way, if you want to write a custom serializer and deserializer and wire that into Jackson, then that would work.